#Practica 3

Vamos a realizar un análisis descriptivo de un dataset sobre los videojuegos más vendidos. Para este análisis tenemos los siguientes datos:

Rank: Posición en el ranking de los videojuegos más vendidos.
Platform: plataforma a la que pertenece el juego.
Year: Año de publicación.
Genre: Género del juego.
Publisher: empresa que comercializa el juego.
NA_Sales: Ventas del juego en América del norte (en millones).
EU_Sales: Ventas del juego en Europa (en millones).
JP_Sales: Ventas del uego en Japon (en millones).
Other_Sales: Ventas del juego en el resto del mundo (en millones).
Global_Sales: Ventas totales del juego (en millones).

Además he creado otro CSV con las diferentes plataformas:
Platform: Plataforma.
Full_Name:Nombre completo de la plataforma.
Company: Empresa a la que pertenece la plataforma.

Cargamos las librerías que vamos a utilizar

In [54]:
import csv
import pandas as pd
import numpy as np

Y cargamos los datos del csv en un dataFrame de pandas

In [55]:
datos = pd.read_csv("data/vgsales.csv")
datos.head(5)

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [56]:
plataforma = pd.read_csv("data/platform.csv")
plataforma.head(5)

Platform                    Full_Name   Company
0     2600                   Atari 2600     Atari
1      3DO  3DO Interactive Multiplayer       3DO
2      3DS                 Nintendo 3DS  Nintendo
3       DC                    DreamCast      Sega
4       DS                  Nintendo DS  Nintendo

In [57]:
datos.shape

(16598, 11)

El fichero original tiene 11 columnas (variables) y 16598 registros.  
Vamos a limpiar el ficheros de datos erroneos y ausentes.

In [58]:
datos=datos[datos['Year']<=2016]
datos=datos[datos['Publisher']!='Unknown']
datos=datos.dropna()

Mostamos un resumen básico de los datos, para hacernos una idea del dataset

In [72]:
datos.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16187.000000  16187.000000  16187.000000  16187.000000  16187.000000   
mean    8265.347254   2006.380923      0.266993      0.148451      0.079131   
std     4786.992191      5.831001      0.824835      0.510773      0.312723   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4114.500000   2003.000000      0.000000      0.000000      0.000000   
50%     8259.000000   2007.000000      0.080000      0.020000      0.000000   
75%    12401.500000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2016.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16187.000000  16187.000000  
mean       0.048675      0.543522  
std        0.190646      1.571897  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.480000  
max       10.570000     82.740000

Mostramos el conteo de videojuegos por plataforma

In [59]:
a=datos.groupby('Platform').count()
a[['Name']]

Name
Platform      
2600       115
3DO          3
3DS        494
DC          52
DS        2108
GB          97
GBA        780
GC         541
GEN         27
GG           1
N64        316
NES         98
NG          12
PC         919
PCFX         1
PS        1187
PS2       2124
PS3       1294
PS4        334
PSP       1190
PSV        402
SAT        173
SCD          6
SNES       239
TG16         2
WS           6
Wii       1278
WiiU       143
X360      1230
XB         803
XOne       212

In [76]:
pd.crosstab(datos.Genre, datos.Platform)

Platform      2600  3DO  3DS  DC   DS  GB  GBA   GC  GEN  GG  ...   SAT  SCD  \
Genre                                                         ...              
Action          54    0  176   3  334   6  161   98    3   0  ...     3    0   
Adventure        1    1   35  11  233   5   36   20    2   0  ...    26    0   
Fighting         1    0   14  12   36   0   23   42    5   0  ...    31    0   
Misc             5    0   53   0  385   8   86   35    1   0  ...    15    2   
Platform         9    0   28   2   88  18  138   73    7   1  ...     5    1   
Puzzle           8    1   19   0  230  15   36   12    0   0  ...     5    0   
Racing           4    0   10   6   66   2   64   60    1   0  ...     8    1   
Role-Playing     0    0   85   4  193  21   73   24    3   0  ...    17    1   
Shooter         22    0    6   3   42   1   40   48    1   0  ...    22    0   
Simulation       1    1   28   1  278   5   18   12    0   0  ...     7    0   
Sports          10    0   25  10  146   9   87  106    3   0  ...    16    0   
Strategy         0    0   15   0   77   7   18   11    1   0  ...    18    1   

Platform      SNES  TG16  WS  Wii  WiiU  X360   XB  XOne  
Genre                                                     
Action          12     0   0  229    63   317  152    68  
Adventure        4     1   0   82     3    47   26    12  
Fighting        25     0   0   41     5    65   44     7  
Misc            17     0   0  269    21   120   45    15  
Platform        26     0   0   56    16    24   49     4  
Puzzle          13     0   0   54     4     6    7     0  
Racing           9     0   0   91     3   102  122    19  
Role-Playing    50     0   4   35     6    74   23    13  
Shooter         10     1   0   59    10   197  124    33  
Simulation       9     0   0   84     1    36   24     3  
Sports          49     0   0  253     8   215  166    35  
Strategy        15     0   2   25     3    27   21     3  

[12 rows x 31 columns]